core/src/test/java/org/elasticsearch/common/io/stream/BytesStreamsTests.java

```diff
@@ -812,4 +812,34 @@ public void testVLong() throws IOException {
         StreamInput in = new BytesArray(Base64.getDecoder().decode("////////////AQAAAAAAAA==")).streamInput();
         assertEquals(-1, in.readVLong());
     }
+
+    public enum TestEnum {
+        ONE,
+        TWO,
+        THREE
+    }
+
+    public void testEnum() throws IOException {
+        TestEnum value = randomFrom(TestEnum.values());
```

Comment:
@StefanSchmidtOz We use a randomization framework that is reproducible (each test execution is associated with a seed).